changelog/changes/itself-large-create.md

```diff
@@ -0,0 +1,50 @@
+---
+title: "HTTP format and compression inference"
+type: feature
+authors: raxyte
+pr: 5300
+---
+
+The `from_http` and `http` operators now automatically infer the file format
+(such as JSON, CSV, Parquet, etc.) and compression type (such as gzip, zstd,
+etc.) directly from the URL's file extension, just like the generic `from`
+operator. This makes it easier to load data from HTTP sources without manually
+specifying the format or decompression step.
+
+If the format or compression cannot be determined from the URL, the operators
+will fall back to using the HTTP `Content-Type` and `Content-Encoding` response
+headers to determine how to parse and decompress the data.
+
+**Examples**
+
+- **Inference Succeeds**
+
+```tql
+from_http "https://example.org/data/events.csv.zst"
+```
+
+The operator infers both the `zstd` compression and the `CSV` format from the
+file extension, decompresses, and parses accordingly.
```

Comment:
Markdown lists don't look great here because the list ends with the code block.

<img width="762" alt="image" src="https://github.com/user-attachments/assets/3e7cf917-e58f-41c0-8f6a-8e753b94b850" />

Reply:
Hmm, let me try numbered list.

Reply:
Just removed the list syntax.